include/nuttx/compiler.h

```diff
@@ -1099,6 +1099,26 @@
 #  undef CONFIG_FS_LARGEFILE
 #endif
 
+/* These are macros to access the current CPU and the current task on a CPU.
```

Comment:
why move sched specific macro to compiler.h

Reply:
The new macro this_cpu_var() depends by nuttx/clock.h, Nesting include will occur if defined in sched.h I retained this_cpu_var in compiler.h and reverted this_cpu() back to sched.h